fetcher.go

```diff
@@ -242,6 +244,9 @@ func (p *pool) fetchResource(ctx context.Context, from string, resource string,
 	}
 
 	req.Header.Add("Accept", mime)
+	if traceId != "" {
+		req.Header.Add("X-Trace-Id", traceId)
```

Comment:
you're happy with this as the header?

Reply:
I switched away from `traceparent` because it has a specified format, and I want the new header to be generic instead of specific to bifrost. Do you have any suggestions?

Reply:
i think this is fine if it's what you want to adopt on the L1 side.

Is there a particular reason to resist the existing traceparent standard / naming?

Reply:
The `traceparent` value has a specified format that the nginx request id doesn't match: https://www.w3.org/TR/trace-context/#version-format. I think it'd be weird if we used a standard name but not the standard value format. 

That being said it might not be a big deal and in that case I would prefer to keep the `traceparent` naming. wdyt?

Reply:
what is the format of the nginx request id?

Reply:
(e.g. is it possible to use that as the 'trace-id' portion of a valid `traceparent` header?

Reply:
nginx is 32 hex chars: `444535f9378a3dfa1b8604bc9e05a303`

the `trace-id` is also 32 hex chars which is great. we'd need another random 16 hex chars for the `parent-id`, the other 2 components `version-format` and `trace-flags` can be hardcoded

Reply:
we can make a span in caboose and generate a valid `parent-id` to go with the `trace-id` for each sub-request

Reply:
that'll also mean that we can differentiate the different 'attempts' that are made within a single bifrost-gateway request context, since they'll have different `parent-id`'s but the same `trace-id`.

Reply:
excellent suggestion, i'll update the PR

Reply:
some issues with this approach:

1. We want to correlate end client requests too, so the `traceparent` header would need to be added to the bifrost-gateway response.
2. `traceparent`s are typically generated by the first caller (arc) and passed down the chain. What we'd be doing instead is generating it at the 2nd caller (technically 3rd if you treat the bifrost lb as a separate caller), then passing it back up and also down the chain. A bit weird but idk if that actually breaks the standard.

Reply:
another thing we can do is have the client add `traceparent` as a query param instead of a header. that would avoid the CORS preflight issue altogether